CommonConfig.mk

```diff
@@ -14,6 +14,9 @@
 
 -include device/sony/customization/Customization.mk
 
+#Broken Rules
+BUILD_BROKEN_DUP_RULES := true
```

Comment:
That's very wrong, the error should be fixed instead.

Reply:
I didn't find the duplicate define of 'vendor.qti.hardware.display.composer-service.xml' in hardware or vendor. So still checking the reason.

Reply:
```
vendor/qcom/opensource/display/sm8150/sdm/libs/hwc2/Android.mk
65:LOCAL_VINTF_FRAGMENTS         := vendor.qti.hardware.display.composer-service.xml

vendor/qcom/opensource/display/sm8250/composer/Android.mk
73:    LOCAL_VINTF_FRAGMENTS         := vendor.qti.hardware.display.composer-service.xml
```

That's where they exist. Maybe they live in `hardware/` in some Q repo? Can you run this in the root of your AOSP tree:

```sh
grep vendor.qti.hardware.display.composer-service.xml device vendor hardware
```
And report the result?

Reply:
```
sjll@buildhost:~/aosp$ grep vendor.qti.hardware.display.composer-service.xml -r device vendor hardware
Binary file vendor/qcom/opensource/display/sm8150/.git/index matches
vendor/qcom/opensource/display/sm8150/sdm/libs/hwc2/Android.mk:LOCAL_VINTF_FRAGMENTS         := vendor.qti.hardware.display.composer-service.xml
sjll@buildhost:~/aosp$ 
```

Reply:
That's correct, I don't really understand why you were seeing this duplicate copy error.

Is it gone now after recent changes?

Reply:
> That's correct, I don't really understand why you were seeing this duplicate copy error.
> 
> Is it gone now after recent changes?

No idea, but this error exists with LOCAL_VINTF_FRAGMENTS. :(

Reply:
Is the makefile where these are defined somehow imported twice, or are you running a tree with more than one display HAL that might provide this fragment? I really don't want to make Q run the inverse of R with https://github.com/sonyxperiadev/hardware-qcom-display/pull/42 :/

Reply:
> Is the makefile where these are defined somehow imported twice, or are you running a tree with more than one display HAL that might provide this fragment? I really don't want to make Q run the inverse of R with [sonyxperiadev/hardware-qcom-display#42](https://github.com/sonyxperiadev/hardware-qcom-display/pull/42) :/

Build system added it twice.
```
FAILED: 
build/make/core/base_rules.mk:526: error: overriding commands for target `out/target/product/pdx201/vendor/etc/vintf/manifest/vendor.qti.hardware.display.composer-service.xml', previously defined at build/make/core/base_rules.mk:526
10:05:35 ckati failed with: exit status 1
```

https://android.googlesource.com/platform/build/+/refs/tags/android-10.0.0_r41/core/base_rules.mk#526

Reply:
Yeah, the Q build system is broken. We don't really even use vintf fragments in Q yet except in GPS, for this reason most likely (Q is ancient enough I forgot it wasn't working properly here :grimacing:).

Since the namespace has to be removed anyway to circumvent:

```
error: vendor/qcom/opensource/display-commonsys-intf/sm8150/Android.bp:5:1: "soong-gralloc_defaults" depends on undefined module "blueprint"
Module "soong-gralloc_defaults" is defined in namespace "vendor/qcom/opensource/display-commonsys-intf/sm8150" which can read these 0 namespaces: []
Module "blueprint" can be found in these namespaces: ["."]
```

Let's go for legacy display Q branches and do it the messy way - as inflicted by AOSP.